### DISCUSSION TITLE
Bot getting banned from hypixel on 1.8 with simple pvp code

### DISCUSSION THREAD

**Discussion by HumanDuck23** [PrismarineJS/mineflayer#2545] [Q&A]
Hi,

I'm trying to make a simple bot for hypixel that plays duels. Here's what I'm doing now:

1. Once the duels game starts, I start a loop that runs this code every 50-250ms to make sure the bot it always looking at the opponent: 
```ts
const e = this.bot.players[this.currentOpponent].entity
this.bot.lookAt(e.position.offset(0, e.height, 0), false)
```
2. Then I have this code in a function for attacking:
```ts
this.attackTimer = setTimeout(() => {
    if (this.currentOpponent && this.bot.players[this.currentOpponent]) {
        const e = this.bot.players[this.currentOpponent].entity
        this.bot.attack(e)
    }
    this.startAttack()
}, TimeUtils.randomMs(150, 300))
```

I start this function in mineflayers `physicsTick` event, if the distance to the opponent is less than 3 blocks and stop it again (just with `clearTimeout`) once the distance is larger than 3 blocks. However, after a few hits, the bot will get banned. Is there something wrong with my code (without the attacking the bot doesn't get banned, it just follows me), or are there packets that aren't being sent by mineflayer that do get sent by a vanilla client?

**Comment by u9g** [comment]
This is not something that is easily testable at all, and there hasn’t been much work put into avoiding anticheats. Nobody can definitively tell you whats causing it, you would have to test a bunch of things

**Comment by aesthetic0001** [comment]
you should probably check the entityAtCursor before attacking

**Reply by HumanDuck23** [reply]
i can't find a method like that, is it documented anywhere?

**Reply by aesthetic0001** [reply]
its in a PR somewhere
